### PR TITLE
chore(in-app): exporting InAppMessage types

### PR DESCRIPTION
**Pull request by esauerbo**
#### Description of changes

Exporting the following types from `react-core-notifications` package 
```
InAppMessage,
InAppMessageAction,
InAppMessageButton,
InAppMessageContent,
InAppMessageImage,
InAppMessageLayout,
InAppMessageStyle,
InAppMessageTextAlign
```
then re-exporting from `react-notifications` and `react-native`

#### Description of how you validated changes
Unit tests passing on affected packages (react-notifications, react-core-notifications, react-native) 
e2e tests passing for in-app 

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -11533,7 +11533,7 @@ cross-spawn@^7.0.0, cross-spawn@^7.0.2, cross-spawn@^7.0.3:
     shebang-command "^2.0.0"
     which "^2.0.1"
 
-crypto-js@^4.0.0:
+crypto-js@^4.0.0, crypto-js@^4.1.1:
```
Comment (calebpollman) [inline yarn.lock:11536]: Looks unrelated to the PR? Same with other _yarn.lock_ change

File: examples/next/pages/ui/components/in-app-messaging/demo/utils.ts [hunk 2]
```diff
@@ -2,10 +2,10 @@ import { useState, useEffect } from 'react';
...
 import {
-  Message,
-  MessageAction,
-  MessageButton,
-  MessageLayout,
+  InAppMessage,
+  InAppMessageAction,
+  InAppMessageButton,
+  InAppMessageLayout,
 } from '@aws-amplify/ui-react-core-notifications';
```
Comment (calebpollman) [inline examples/next/pages/ui/components/in-app-messaging/demo/utils.ts:9]: This import should be from '@aws-amplify/ui-react-notifications' if we are exposing the types?

File: packages/react-notifications/src/index.ts [hunk 3]
```diff
@@ -1 +1,11 @@
 export * from './components/InAppMessaging';
+export {
```
Comment (calebpollman) [inline packages/react-notifications/src/index.ts:2]: Let's add a comment here:
```suggestion
// re-export in-app types from '@aws-amplify/ui-react-core-notifications'
export {
```

and move the re-export in to  _./components/InAppMessaging_

**Review by thaddmt — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]
